File: Easy/Second Largest/second-largest.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;

// } Driver Code Ends
//User function template for C++
class Solution{
public:	
	// Function returns the second
	// largest elements
    int print2largest(int arr[], int n) {
        int maxElement = INT_MIN;
        int secondMaxElement = INT_MIN;
        for(int i = 0; i< n ; i++){
            if(arr[i] > maxElement){
                secondMaxElement = maxElement;
                maxElement = arr[i];
            } else if(arr[i] != maxElement && arr[i] > secondMaxElement){
                secondMaxElement = arr[i];
            }
        }
        return secondMaxElement != INT_MIN ? secondMaxElement : -1;
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr[n];
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        auto ans = ob.print2largest(arr, n);
        cout << ans << "\n";
    }
    return 0;
}

// } Driver Code Ends

File: Easy/Second Largest/second-largest.py
#User function Template for python3
class Solution:
	def print2largest(self,arr, n):
		# code here
        max_element = float('-inf')
        second_max_element = float('-inf')
        for num in arr:
            if num>max_element:
                second_max_element = max_element;
                max_element = num;
            elif num!=max_element and num >second_max_element:
                second_max_element = num
        return second_max_element if second_max_element != float('-inf') else -1

#{ 
 # Driver Code Starts
# Initial Template for Python 3
if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n = int(input())
        arr = list(map(int, input().strip().split()))
        ob = Solution()
        ans = ob.print2largest(arr, n)
        print(ans)
        tc -= 1

# } Driver Code Ends